src/dmd/parse.d

```diff
@@ -6029,19 +6029,56 @@ final class Parser(AST) : Lexer
         switch (token.value)
         {
         case TOK.leftCurly:
-            /* Scan ahead to see if it is a struct initializer or
-             * a function literal.
-             * If it contains a ';', it is a function literal.
-             * Treat { } as a struct initializer.
+            /* Scan ahead to discern between struct initializer and
+             * parameterless function literal.
+             *
+             * We'll scan the topmost curly bracket level for statement-related
+             * tokens, thereby ruling out a struct initializer.  (A struct
+             * initializer which itself contains function literals may have
+             * statements at nested curly bracket levels.)
+             *
+             * It's important that this function literal check not be
```

Comment:
So I was trying to say that this should probably be e.g. "check isn't" as "not be" sounds really strange here and I'm not sure that actually works with English grammar.

Reply:
I believe the grammar is correct either way.  Pondering why I worded it this way, it was to avoid the uncomfortable pattern "it is [...] is not".  E.g.:
>     "It is important that this elephant is not in the room." vs.
>     "It is important that this elephant not be in the room."